### PR TITLE
mlhka: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new package mlhka. In install phase, installing the binary only works if I specify `with working_dir(self.stage.source_path)`. If that's not specified I get a file not found error

### CODE REVIEW

File: var/spack/repos/builtin/packages/mlhka/package.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    homepage = "https://wright.eeb.utoronto.ca"
+    url      = "https://github.com/rossibarra/MLHKA"
+
+    version('2.1', git='https://github.com/rossibarra/MLHKA.git',
+            commit='e735ddd39073af58da21b00b27dea203736e5467')
+
+    def patch(self):
+        command_line = (spack_cxx, 'MLHKA_version{0}.cpp'.format(self.version),
+                        '-o', 'MLHKA')
+        subprocess.Popen(command_line)
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/mlhka/package.py]: In a `Package`, this stuff should go in install -- `patch()` isn't supposed to actually build anything (just prepare the source).  Can you move it?

This is probably more idiomatic:

```python
cxx = which('c++')
cxx('MLHKA_version{0}.cpp'.format(self.version), '-o', 'MLHKA')
```

The spack compiler wrappers are available *both* as specifically named things (e.g., `gcc`, `icc`, depending on what you're using), *and* their generic names (for situations like this, where the build system or lack thereof is not sensitive to the compiler name, and you just know you need the `cc` or `c++` compiler.

**Comment by tgamblin** [comment]
> installing the binary only works if I specify with working_dir(self.stage.source_path). If that's not specified I get a file not found error

What happens if you `print os.getcwd()` from inside `install()`?  Can you attach your build.out?

**Comment by akthoma** [comment]
@tgamblin It no longer has the issue after removing the `patch` phase and making the `c++` change. Thanks!

**Comment by tgamblin** [comment]
still needs a `flake8` fix -- you can run that locally with `spack flake8` (assuming you have `flake8` installed.

**Comment by tgamblin** [comment]
@akthoma: i pushed the flake8 fix for you

**Comment by akthoma** [comment]
@tgamblin thank you!

**Pull request merged**